services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -803,20 +804,18 @@ public static IdentityCookieToken createIdentityToken(KeycloakSession keycloakSe
     }
 
     public static void createLoginCookie(KeycloakSession keycloakSession, RealmModel realm, UserModel user, UserSessionModel session, UriInfo uriInfo, ClientConnection connection) {
+        assert session != null : "User session cannot be null";
```

Comment:
I was mistaken about `assert`, it seems that it requires a runtime flag for it to work properly, and is therefore not really a valid option. We could use `org.wildfly.common.Assert` here though:

```java
Assert.checkNotNullParam("session", session);
```

Alternatively, since this is such a common case and we likely don't want to further depend on anything WildFly we just use the same old null check.


```java
if (session == null) {
    throw new IllegalArgumentException("The 'session' parameter cannot be null.");
}
```

Reply:
Ok, I've replaced it with `Objects.requireNonNull` . Not really big deal as `session` passed to this method is never null currently, so this is more convenient just for the future if someone uses this method in the future.